server/src/__tests__/analyzer.test.ts

```diff
@@ -107,3 +111,200 @@ describe('analyze', () => {
     expect(word2).toEqual('BAR')
   })
 })
+
+describe('getDirectiveFileUris', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('gets directive file URIs', async () => {
+    const parsedBarPath = path.parse(FIXTURE_DOCUMENT.BAR_INC.uri.replace('file://', ''))
+    const parsedFooPath = path.parse(FIXTURE_DOCUMENT.FOO_INC.uri.replace('file://', ''))
+    const parsedBazPath = path.parse(FIXTURE_DOCUMENT.BAZ_BBCLASS.uri.replace('file://', ''))
+
+    jest.spyOn(bitBakeProjectScanner, 'includes', 'get').mockReturnValue([
+      {
+        name: parsedBarPath.name,
+        path: parsedBarPath,
+        extraInfo: 'layer: core'
+      },
+      {
+        name: parsedFooPath.name,
+        path: parsedFooPath,
+        extraInfo: 'layer: core'
+      }
+    ])
+
+    jest.spyOn(bitBakeProjectScanner, 'classes', 'get').mockReturnValue([{
+      name: parsedBazPath.name,
+      path: parsedBazPath,
+      extraInfo: 'layer: core'
+    }])
+
+    const parser = await generateParser()
+    const analyzer = await getAnalyzer()
+
+    const parsedTree = parser.parse(FIXTURE_DOCUMENT.DIRECTIVE.getText())
+    const fileUris = analyzer.getDirectiveFileUris(parsedTree)
+
+    expect(fileUris).toEqual(
+      expect.arrayContaining([
+        FIXTURE_URI.BAR_INC,
+        FIXTURE_URI.FOO_INC,
+        FIXTURE_URI.BAZ_BBCLASS
+      ])
+    )
+  })
+})
+describe('sourceIncludeFiles', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('finds and analyzes files in the directive statements for the first time', async () => {
+    const analyzer = await getAnalyzer()
+    const uri = FIXTURE_URI.DIRECTIVE
+
+    jest.spyOn(Analyzer.prototype, 'getDirectiveFileUris').mockReturnValueOnce([
+      FIXTURE_URI.BAR_INC,
+      FIXTURE_URI.FOO_INC,
+      FIXTURE_URI.BAZ_BBCLASS
+    ])
+
+    const fsReadFileSyncMock = jest.spyOn(fs, 'readFileSync')
+
+    analyzer.sourceIncludeFiles(uri, [])
+
+    expect(fsReadFileSyncMock).toHaveBeenCalledWith(FIXTURE_URI.DIRECTIVE.replace('file://', ''), 'utf8')
+    expect(fsReadFileSyncMock).toHaveBeenCalledWith(FIXTURE_URI.FOO_INC.replace('file://', ''), 'utf8')
+    expect(fsReadFileSyncMock).toHaveBeenCalledWith(FIXTURE_URI.BAR_INC.replace('file://', ''), 'utf8')
+    expect(fsReadFileSyncMock).toHaveBeenCalledWith(FIXTURE_URI.BAR_INC.replace('file://', ''), 'utf8')
+
+    expect(analyzer.getAnalyzedDocument(FIXTURE_URI.FOO_INC)).not.toBeUndefined()
+    expect(analyzer.getAnalyzedDocument(FIXTURE_URI.BAR_INC)).not.toBeUndefined()
+    expect(analyzer.getAnalyzedDocument(FIXTURE_URI.BAZ_BBCLASS)).not.toBeUndefined()
+  })
+
+  it('does not read the files and parse the syntax tree when the documents were already analyzed ', async () => {
```

Comment:
Nice!

Reply:
Thanks, but just in case, this test is for the analyze operations (not entirely the same as `analyze()`) that are happening in the sourceIncludeFiles()